### PR TITLE
dot escape

### PR DESCRIPTION
**Pull request by boriwo**

### CODE REVIEW

File: internal/pkg/app/handlers_v1.go [hunk 1]
```diff
@@ -58,23 +63,86 @@ type APIManager struct {
...
 	addRouteSuccessRecorder    metric.BoundFloat64Counter
 	addRouteFailureRecorder    metric.BoundFloat64Counter
 	removeRouteSuccessRecorder metric.BoundFloat64Counter
 	removeRouteFailureRecorder metric.BoundFloat64Counter
+	globalWebhookOrg           string
+	globalWebhookApp           string
+	globalWebhookRouteId       string
+}
+
+type CachedTenantConfig struct {
```
Comment (holidaymike) [inline internal/pkg/app/handlers_v1.go:76]: What do you think about using a 3rd party cache library: https://pkg.go.dev/github.com/ReneKroon/ttlcache/v2

File: internal/pkg/app/handlers_v1.go [hunk 2]
```diff
@@ -204,13 +326,19 @@ func (a *APIManager) sendEventHandler(w http.ResponseWriter, r *http.Request) {
 		resp.Respond(ctx, w, doYaml(r))
 		return
 	}
-	tenantConfig, err := a.tenantStorer.GetConfig(ctx, *tid)
-	if err != nil {
-		log.Ctx(ctx).Error().Str("op", "sendEventHandler").Str("error", err.Error()).Msg("error getting tenant config")
-		resp := ErrorResponse(convertToApiError(ctx, err))
-		resp.Respond(ctx, w, doYaml(r))
-		return
+	tenantConfig := a.tenantCache.GetTenant(tid.Key())
```
Comment (holidaymike) [inline internal/pkg/app/handlers_v1.go:329]: Should we mutex lock this section such that if there are tons of requests, they don't all update the cache at the same time? Alternatively, if we don't want to lock, maybe we can return the expired tenantConfig and kickoff a go routine to update cache in the background.

File: internal/pkg/app/middlewares.go [hunk 3]
```diff
@@ -85,7 +85,11 @@ func authenticateMiddleware(next http.Handler) http.Handler {
 			return
 		}
 		var tid *tenant.Id
-		if strings.HasPrefix(r.URL.Path, "/ears/v1/routes") ||
+		if strings.HasPrefix(r.URL.Path, "/ears/v1/events") {
```
[the remark this answers is not in the document]
Comment (boriwo) [reply internal/pkg/app/middlewares.go:88]: yes, i think you are right, will fix

File: internal/pkg/app/handlers_v1.go (hunk 2 above)
Comment (boriwo) [reply internal/pkg/app/handlers_v1.go:329]: ok

File: internal/pkg/app/handlers_v1.go (hunk 1 above)
Comment (boriwo) [reply internal/pkg/app/handlers_v1.go:76]: haha, you know how i feel about using other people's code...

File: internal/pkg/app/handlers_v1.go (hunk 2 above)
Comment (boriwo) [reply internal/pkg/app/handlers_v1.go:337]: i protected this with locks for now fro thread safety, i'm not sure ur other suggestions would really help with the performance issue you identified. i suggest we rerun your load test on the current version of the code and see how it goes

File: internal/pkg/app/middlewares.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (boriwo) [reply internal/pkg/app/middlewares.go]: i also left the /ears path in there as an alternative - might come in handy at some point

File: internal/pkg/app/handlers_v1.go (hunk 2 above)
Comment (holidaymike) [reply internal/pkg/app/handlers_v1.go:337]: For my comment, I was saying that we should lock line 337 to line 348 (although locking the cache map make sense too :)

**Review by holidaymike — APPROVED** [review]

**Pull request merged**